TestDatetimeIndex.test_reindex_with_same_tz 32-bit CI failing in MacPython

https://dev.azure.com/pandas-dev/pandas-wheels/_build/results?buildId=34375&view=logs&j=517fe804-fa30-5dc2-1413-330699242c05&t=2e128ad5-2f7f-5333-3f34-c85b8fbc7250&l=1267

```
=================================== FAILURES ===================================
_________________ TestDatetimeIndex.test_reindex_with_same_tz __________________
[gw1] linux -- Python 3.7.0 /venv/bin/python

        rng_a = date_range("2010-01-01", "2010-01-02", periods=24, tz="utc")
        rng_b = date_range("2010-01-01", "2010-01-02", periods=23, tz="utc")
        result1, result2 = rng_a.reindex(
            rng_b, method="nearest", tolerance=timedelta(seconds=20)
        )
        expected_list1 = [
            "2010-01-01 00:00:00",
            "2010-01-01 01:05:27.272727272",
            "2010-01-01 02:10:54.545454545",
            "2010-01-01 03:16:21.818181818",
            "2010-01-01 04:21:49.090909090",
            "2010-01-01 05:27:16.363636363",
            "2010-01-01 06:32:43.636363636",
            "2010-01-01 07:38:10.909090909",
            "2010-01-01 08:43:38.181818181",
            "2010-01-01 09:49:05.454545454",
            "2010-01-01 10:54:32.727272727",
            "2010-01-01 12:00:00",
            "2010-01-01 13:05:27.272727272",
            "2010-01-01 14:10:54.545454545",
            "2010-01-01 15:16:21.818181818",
            "2010-01-01 16:21:49.090909090",
            "2010-01-01 17:27:16.363636363",
            "2010-01-01 18:32:43.636363636",
            "2010-01-01 19:38:10.909090909",
            "2010-01-01 20:43:38.181818181",
            "2010-01-01 21:49:05.454545454",
            "2010-01-01 22:54:32.727272727",
            "2010-01-02 00:00:00",
        ]
        expected1 = DatetimeIndex(
            expected_list1, dtype="datetime64[ns, UTC]", freq=None,
        )
        expected2 = np.array([0] + [-1] * 21 + [23], dtype=np.int64,)
        tm.assert_index_equal(result1, expected1)
>       tm.assert_numpy_array_equal(result2, expected2)
E       AssertionError: numpy array are different
E       
E       Attribute "dtype" are different
E       [left]:  int32
E       [right]: int64

```

--- a/pandas/tests/indexes/datetimes/test_datetime.py
+++ b/pandas/tests/indexes/datetimes/test_datetime.py
@@ -53,7 +53,7 @@ class TestDatetimeIndex:
         expected1 = DatetimeIndex(
             expected_list1, dtype="datetime64[ns, UTC]", freq=None,
         )
-        expected2 = np.array([0] + [-1] * 21 + [23], dtype=np.int64,)
+        expected2 = np.array([0] + [-1] * 21 + [23], dtype=np.dtype("int"))
         tm.assert_index_equal(result1, expected1)
         tm.assert_numpy_array_equal(result2, expected2)
 

--- a/pandas/tests/indexes/datetimes/test_datetime.py
+++ b/pandas/tests/indexes/datetimes/test_datetime.py
@@ -53,7 +53,7 @@ class TestDatetimeIndex:
         expected1 = DatetimeIndex(
             expected_list1, dtype="datetime64[ns, UTC]", freq=None,
         )
-        expected2 = np.array([0] + [-1] * 21 + [23], dtype=np.dtype("int"))
+        expected2 = np.array([0] + [-1] * 21 + [23], dtype=np.dtype("intp"))
         tm.assert_index_equal(result1, expected1)
         tm.assert_numpy_array_equal(result2, expected2)
 

